main.permission_bindings.tf

```diff
@@ -0,0 +1,16 @@
+module "permission_bindings" {
+  source   = "./modules/permission_binding"
+  for_each = var.permission_bindings
+
+  client_group_name = each.value.client_group_name
```

Comment:
It should be possible to build an implicit dependency here by referencing `module.client_groups`